litmus-portal/graphql-server/pkg/k8s/cluster.go

```diff
@@ -94,11 +93,56 @@ func CreateDeployment(namespace, token string) (*appsv1.Deployment, error) {
 	return result, nil
 }
 
-func GetKubeConfig() (*rest.Config, error) {
-	if kubeConfig := os.Getenv("KUBECONFIG"); kubeConfig != "" {
-		return clientcmd.BuildConfigFromFlags("", kubeConfig)
+func GetPortalEndpoint() (string, error) {
+	var (
+		Nodeport       int32
+		ExternalIP     string
+		InternalIP     string
+		LitmusPortalNS = os.Getenv("LITMUS_PORTAL_NAMESPACE")
+	)
+
+	clientset, err := GetGenericK8sClient()
+	if err != nil {
+		log.Print("Error from here")
+		return "", err
+	}
+
+	podList, _ := clientset.CoreV1().Pods(LitmusPortalNS).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "component=litmusportal-server",
+	})
+
+	svc, err := clientset.CoreV1().Services(LitmusPortalNS).Get(context.TODO(), "litmusportal-server-service", metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	fmt.Print(svc)
+	fmt.Print("here0")
+	fmt.Print(svc.Spec)
```

Comment:
Remove `fmt` and use `log` with proper message for logging

Reply:
done